# 1. 
The key factor that makes the ideas that can be examined and tested statistically is that the ideas are based on objective datas. For those ideas that cannot be tested statistically, they are mostly based on subjective opinions, which could not be datamined.
A good null hypothesis is that it could be suppoeted by large amount of data, which means the evidence is convincing. 
Null hypothesis is to assume "Nothing happened". An althernative hypothesis is to assume "Something happened". Null hypothesis and althernative hypothesis are contradicted with each other. 


# 2.
𝑥𝑖's (xi): These are individual data points or observations in a dataset, which is picked randomly. For example, in a sample of heights, each 𝑥𝑖 represents the height of a single person.
𝑥¯ (x-bar): This symbol represents the sample mean. It's calculated by summing all 𝑥𝑖's which is ramdonly picked from the sample and dividing by the number of observations 𝑛. 𝑥¯ gives us an estimate of the population mean 𝜇 based on the sample.
𝜇 (mu): This symbol represents the population mean. It's the average value of the entire population, which is often unknown and estimated using 𝑥¯ from a sample.
𝜇0 (mu-zero): This typically represents a hypothesized population mean in statistical hypothesis testing. For instance, in a hypothesis test comparing sample means, 𝜇0 could be the mean under the null hypothesis (the hypothesis that assumes no effect or no difference).

chat summary:
https://chatgpt.com/share/6706b785-8d0c-8008-b7d9-03c44304fa15

# 3.
We need to assume H0 is true, because we need the assumption of H0 to calculate p-value. As null hypothesis is contradicted with althernative hypothesis, the only way to say althernative hypothesis is true is to reject null hypothesis. To reject null hypothesis, we need have a big p-value to provide enough evidence. By assuming null hypothesis is true, we could calculate the p-value. Then, we could have evidence to prove whether null hypothesis could be rejected or not. 


# 4.
From the example of the video, we assume that the dog Fido is innocent. So the crime of having gotten into the garbage bin should not be caused by Fido but other reasons, such as our neighbour breaks into our house. However, the possibility of the hypothesis "neighbour breaks into our house" is too small (p-value), rejecting the assumption "Fido is innocent". The possibility of the hypothesis "neighbour breaks into our house" means the p-value. As the p-value is small, we could assure null hypothesis could be rejected.  

# 5.
H0: 50% of the kissing couples tilt their heads
HA: 50% of the kissing couples don't tilt their heads

In [1]:
import numpy as np

observed_right_tilts = 80
total_couples = 124
simulations = 10000 

simulated_tilts = np.random.binomial(n=total_couples, p=0.5, size=simulations)
p_value = np.mean(simulated_tilts >= observed_right_tilts)

print(f"Simulated p-value: {p_value}")

Simulated p-value: 0.001


According to the table, the simulated p-value 0.001 provides a strong evidence that the H0 could be rejected, meaning 50% of the kissing couples don't tilt their heads.

# 6.
We should define how small that p-value is. For example, if p-value <= 0.001, this shows a very strong evidence that the null hypothesis is false. if 0.001 < p-value < 0.01, this shows relatively strong evidence that the null hypothesis is false. if 0.01 < p-value <= 0.05, this shows a moderate evidence that the null hypothesis is false. We need a range for us to evaluate whether the p-value is small enough to provide evidence of rejecting null hypothesis. 
Theoretically, we can't have any definite conclusion. All p-value we have is a possiblity not a "True" or "False" choice. We just have the range of the p-value providing evidence to enable us make evaluation: whether we have enough evidence to reject null hypothesis. There is no definite cases. 

# 7.


In [3]:
# Create the HealthScoreChange column
patient_data['HealthScoreChange'] = patient_data['FinalHealthScore'] - patient_data['InitialHealthScore']

# Now proceed with the hypothesis test calculations
import scipy.stats as stats

# Calculate the mean and standard deviation of the HealthScoreChange
mean_change = patient_data['HealthScoreChange'].mean()
std_change = patient_data['HealthScoreChange'].std(ddof=1)
n = len(patient_data)

# Calculate the t-statistic for one-sided test (null hypothesis: no change or FinalHealthScore <= InitialHealthScore)
t_statistic = mean_change / (std_change / n**0.5)

# Compute the one-sided p-value (alternative hypothesis: FinalHealthScore > InitialHealthScore)
p_value = 1 - stats.t.cdf(t_statistic, df=n-1)

print(f"T-statistic: {t_statistic}")
print(f"One-sided p-value: {p_value}")


T-statistic: 2.5900615612703937
One-sided p-value: 0.014605261907746248


We could expect one-side p-value is smaller than "two tailed" p-value, as it only evaluate one side of test. One-side value focuses on deviations in only one direction (either greater than or less than a certain value). It considers just one extreme tail of the distribution.

Chat summary:
https://chatgpt.com/share/6706d4e2-3aac-8008-8b47-7366f7a98f72

# 8.
H0 = 50% of people could tell the difference.
HA = more than 50% of people could tell the difference.

In [13]:
import numpy as np

differentiated = 49
total = 80
simulations = 80

np.random.seed(10000)

p_values = []

for _ in range(simulations):
    simulated_differentiated = np.random.binomial(n=total, p=0.5)
    p_values.append(simulated_differentiated >= differentiated)

p_value = np.mean(p_values)

print(f"Simulated p-value: {p_value}")


Simulated p-value: 0.0375


p-value = 0.0375 0.05 >= p > 0.01 showing a moderate evidence against the null hypothesis

# 9. No